app/forms/provider_interface/provider_relationship_permissions_setup_wizard.rb

```diff
@@ -73,26 +73,24 @@ def permissions_for_persistence
     # 3. check
     def next_step
       if checking_answers.present?
-        [:check]
-      elsif current_step == 'organisations'
-        [:permissions, next_provider_relationship_id]
-      elsif current_step == 'permissions' && next_provider_relationship_id.present?
-        [:permissions, next_provider_relationship_id]
-      else
-        [:check]
+        return [:check]
+      elsif current_step == 'organisations' || (current_step == 'permissions' && next_provider_relationship_id.present?)
+        return [:permissions, next_provider_relationship_id]
       end
+
+      [:check]
```

Comment:
This whole bit could be:
```
if current_step == 'organisations' || (current_step == 'permissions' && next_provider_relationship_id.present?)
  [:permissions, next_provider_relationship_id]
else
  [:check]
end
```

Reply:
`checking_answers` can be set to true at any step of the process, so that's why we can't switch to that statement.

Reply:
Ah yeah I see. I don't think writing it in this way is the intention of the cop.

Could we extract variables and write the conditions like this:
```
next_step_is_permissions = current_step == 'organisations' || (current_step == 'permissions' && next_provider_relationship_id.present?)

if next_step_is_permissions && !checking_answers
  [:permissions, next_provider_relationship_id]
else
  [:check]
end
```

Reply:
Good idea. Maybe extract the permissions check is another method